src/sleep/sleep.c

```diff
@@ -390,7 +390,12 @@ static int custom_timer_suspend(const SleepConfig *sleep_config) {
                         }
                 }
 
-                /* Do not suspend more than HibernateDelaySec= */
+                /* Do not suspend more than HibernateDelaySec= unless HibernateOnACPower=no and currently on AC power */
+
+                if (!sleep_config->hibernate_on_ac_power && on_ac_power()) {
+                    hibernate_timestamp = usec_add(now(CLOCK_BOOTTIME), sleep_config->hibernate_delay_usec);
```

Comment:
Hmm, so I think this option makes sense, but should probably be effective only if the system actually has battery. I.e. let's not allow s2h to "decay" to suspend.